Operation framework doc updates (#3833)
LinuxForHealth/FHIR: docs/src/pages/guides/FHIROperationFramework.md

---
layout: default
title: Extended Operations Framework
date:  2022-08-02
---

# Extended Operations Framework

> The HL7 FHIR Specification describes a [framework](https://hl7.org/fhir/R4B/operations.html) for operations that go beyond the strandard REST API.

Author: ```suggestion
The HL7 FHIR Specification describes a [framework](https://hl7.org/fhir/R4B/operations.html) for operations that go beyond the standard REST API.
```

  - Be sure to have a return parameter
  - Typically this is a FHIR Resource

# The LinuxForHealth Extended Operations Framework implementation

The LinuxForHealth FHIR Server has integrated the Extended Operations Framework into the code base.

![Framework](https://raw.githubusercontent.com/wiki/LinuxForHealth/FHIR/operation/operation-framework.png)

> Upon startup of the LinuxForHealth FHIR Server webapplication, the FHIR Operation Registry uses the Java ServiceLoader framework. Each Operation is registered with the FHIROperationRegistry using `META-INF/services/com.ibm.fhir.server.operation.spi.FHIROperation` which lists each of the classes that implement the FHIROperation interface

Author: ```suggestion
On startup, the FHIR Operation Registry uses the Java ServiceLoader framework. Each Operation is registered with the FHIROperationRegistry using `META-INF/services/com.ibm.fhir.server.operation.spi.FHIROperation` which lists each of the classes that implement the FHIROperation interface
```

# Questions and Support
If you need further support, you can reach out to the development team at:

> - [Zulip: LinuxForHealth]([https://chat.fhir.org/#narrow/stream/212434-ibm](https://chat.fhir.org/#narrow/stream/212434-LinuxForHealth))

Author: ```suggestion
- [Zulip: LinuxForHealth](https://chat.fhir.org/#narrow/stream/212434-LinuxForHealth)
```
